components/skus/browser/rs/lib/src/sdk/credentials/mod.rs

```diff
@@ -122,6 +133,36 @@ where
         }))
     }
 
+    #[instrument]
+    pub async fn next_active_at(&self, item_id: &str) -> Result<Option<NaiveDateTime>, SkusError> {
+        let now = Utc::now().naive_utc();
+        let creds = self.client.get_time_limited_v2_creds(item_id).await?;
+
+        match creds {
+            Some(tlv2_creds) => {
+                // Check if unblinded_creds is present and filter out unspent credentials with
+                // valid_from greater than now
+                let next_valid_from = tlv2_creds
+                    .unblinded_creds
+                    .unwrap_or_default()
+                    .into_iter()
+                    .filter_map(|tlv2_cred| {
+                        tlv2_cred
+                            .unblinded_creds
+                            .unwrap_or_default()
+                            .into_iter()
+                            .filter(|single_cred| !single_cred.spent && tlv2_cred.valid_from > now)
+                            .map(|_| tlv2_cred.valid_from)
+                            .next()
```

Comment:
Sorry, this is a little mazey with the double loop. What is the `next()` doing here? If the goal is to find a the youngest, shouldn't you flatten and look at all the future `SingleUseCredential`s? If `unblinded_creds` is sorted, a comment to that effect would help.

Doc comments on the wrapped wrapper types explaining the data flow would also help, I think.

Reply:
Ok, thanks for explaining. That makes more sense; I hadn't noticed that `valid_from` covers the whole container. Just expanding your comment to include what you said here would be good, I think.

Reply:
Updated @rillian